cloudant-sync-datastore-android/src/main/java/com/cloudant/sync/replication/PeriodicReplicationService.java

```diff
@@ -156,19 +164,29 @@ public ServiceHandler(Looper looper) {
 
         @Override
         public void handleMessage(Message msg) {
+            Intent intent = msg.getData().getParcelable(EXTRA_INTENT);
+
             switch (msg.arg2) {
                 case COMMAND_START_PERIODIC_REPLICATION:
                     startPeriodicReplication();
+                    releaseWakeLock(intent);
```

Comment:
On the face of it, that sounds like a nice idea, but we don't want to call `releaseWakeLock` or `stopSelf` in the `default` case, as the processing gets passed on to the `ReplicationService.ServiceHandler` in the superclass. There, when we handle a `COMMAND_START_REPLICATION` message, we don't want to release the wake lock or stop the service when we've processed the message in the `handleMessage()` method.  We need to wait until the replications have actually completed and the `allReplicationsCompleted()` callback is executed before we release the wake lock and stop the service.

I think even if we move the call to `super.handleMessage(msg)` to the `default`, using a `finally` block is problematic as if we did likewise in the superclass we'd end up attempting to release the wake lock and stop the service twice for the cases handled by the superclass.

I can't think of a clean way of doing it using `finally`.